internal/wallet/wallet.go

```diff
@@ -6,35 +6,50 @@ import (
 	"path/filepath"
 	"time"
 
+	"github.com/adiabat/btcd/btcec"
 	"github.com/btcsuite/btcd/chaincfg"
 	"github.com/btcsuite/btcwallet/waddrmgr"
 	"github.com/btcsuite/btcwallet/walletdb"
 	_ "github.com/btcsuite/btcwallet/walletdb/bdb" // blank import for bolt db driver
 	"github.com/btcsuite/btcwallet/wtxmgr"
 )
 
+// Wallet is an interface that provides access to manage pubkey addresses and
+// sign scripts of managed addressesc using private key. It also manags utxos.
+type Wallet interface {
+	CreateAccount(
+		scope waddrmgr.KeyScope, name string, privPass []byte,
+	) (account uint32, err error)
+
+	NewPubkey() (*btcec.PublicKey, error)
```

Comment:
I expect `btcec.PublicKey` instead of `waddrmgr.ManagedPubKeyAddress` because I want wallet to hide its implementation.